### PR TITLE
Homogenize with dcos/dcos-go

### PR DESCRIPTION
**Pull request by malnick**
- Use dcos-go where possible
  - detect IP
  - detect mesos ID
  - detect cluster ID
- Use the dcos client if IAM config is present for dcos-go lookups (specifically Mesos ID)

### CODE REVIEW

File: collector/framework_collector.go [hunk 1]
```diff
@@ -50,15 +50,18 @@ type AvroDatum struct {
...
 		tagData       = avroRecord{}
 		datapointData = avroRecord{}
 		// Initialize the metrics message
 		pmm = producers.MetricsMessage{
 			Name: "dcos.metrics.app",
 			Dimensions: producers.Dimensions{
-				Labels: make(map[string]string),
+				ClusterID: clusterID,
+				MesosID:   mesosID,
+				Hostname:  ipaddress,
```
Comment (darkonie) [inline collector/framework_collector.go]: Hostname: ipaddress?

Comment (malnick) [inline collector/framework_collector.go]: Yolo.

Comment (malnick) [inline collector/framework_collector.go]: I'll add some hostname code later, just defaulting it to this for now.

**Comment by darkonie** [comment]
👍

**Pull request merged**